src/ScalarMessaging.js

```diff
@@ -286,6 +286,30 @@ function inviteUser(event, roomId, userId) {
     });
 }
 
+/**
+ * Returns a promise that resolves when a widget with the given
+ * ID has been added as a user widget (ie. the accountData event
+ * arrives) or rejects after a timeout
+ */
+function waitForUserWidget(widgetId) {
+    return new Promise((resolve, reject) => {
+        let timerId;
+        function onAccountData(ev) {
+            if (ev.getContent()[widgetId] !== undefined) {
+                MatrixClientPeg.get().removeListener('accountData', onAccountData);
+                clearTimeout(timerId);
+                resolve();
+            }
+        }
+        timerId = setTimeout(() => {
+            console.log("Timed out waiting for widget ID " + widgetId + " to appear");
+            MatrixClientPeg.get().removeListener('accountData', onAccountData);
+            reject();
+        }, 10000);
+        MatrixClientPeg.get().on('accountData', onAccountData);
```

Comment:
We need to handle the case where the account data has already arrived (and resolve immediately).

Reply:
Ah yep, I was doing this by creating the promise before sending the event, but this is neater.

Reply:
Oh right, so you were. Agreed that it's neater, clearer.